connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -181,7 +184,11 @@ public void execute() {
      */
     private boolean sendRecords() {
         int processed = 0;
-        for (final SourceRecord record : toSend) {
+        for (SourceRecord preTransformRecord : toSend) {
+            final SourceRecord record = transformationChain.apply(preTransformRecord);
+
+            if (record == null) continue;
```

Comment:
I think invoking `commitRecord` anyway is important. It seems like connectors relying on overriding `commitRecord` will just break. I get that it is kind of misleading, but once it is discarded we can effectively ack it for the source system. I think it'd make sense to document this, but I think it should just be a warning note in the javadocs for `commitRecord`.